### PR TITLE
Improve error handling in JS adapter's updateToken

### LINKED ISSUE #13852: js adapter just sets error to true upon error updateToken
### Describe the bug

If an error occurs while refreshing a token, the error returned is just true, instead of the actual error

The problem is in the updateToken function in https://github.com/keycloak/keycloak/blob/18.0.1/adapters/oidc/js/src/keycloak.js line 661

### Version

18

### Expected behavior

Promise is rejected with the reason true no matter what the cause of the error is

### Actual behavior

That the promise is rejected with an error message that relates to the actual error. Perhaps simply the http code (and perhaps response text) when available

The thing is, that if we had the info, we could for example retry refreshing the token in cases where the server is down or internet connectivity is lost.

### How to Reproduce?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by adropofliquid**
Closes #13852

### CODE REVIEW

File: js/libs/keycloak-js/lib/keycloak.js [hunk 1]
```diff
@@ -673,7 +673,7 @@ function Keycloak (config) {
 
                                 kc.onAuthRefreshError && kc.onAuthRefreshError();
                                 for (var p = refreshQueue.pop(); p != null; p = refreshQueue.pop()) {
-                                    p.setError(true);
+                                    p.setError("Failed to refresh the token. Status: " + req.status);
```
Comment (jonkoops) [inline js/libs/keycloak-js/lib/keycloak.js:676]: Let's not add any status codes or network details here, that is what the network inspector for the devtools are for.
```suggestion
                                    p.setError("Failed to refresh token: An unexpected HTTP error occurred while attempting to refresh the token.");
```

**Review by jonkoops — CHANGES_REQUESTED** [review]

**Comment by jonkoops** [comment]
PR looks good to me like this. @adropofliquid could you squash your commits and [sign off](https://github.com/keycloak/keycloak/pull/34113/checks?check_run_id=31778712387) on your commit?

**Comment by adropofliquid** [comment]
@jonkoops  just confirming I have completed what you asked. Anything else?

**Review by jonkoops — APPROVED** [review]
LGTM

**Comment by adropofliquid** [comment]
I'm also seeing that this branch is out of date and won't auto-merge:

This branch is out-of-date with the base branch
Merge the latest changes from main into this branch.

**Comment by jonkoops** [comment]
It will, there is no need to rebase unless there are conflicts.

**Pull request merged**